src/keygen/keyshare.rs

```diff
@@ -17,19 +22,23 @@ use zeroize::ZeroizeOnDrop;
 ///
 /// # 🔒 Storage requirements
 /// This type must be stored securely by the calling application.
-#[derive(Clone, ZeroizeOnDrop)]
+#[derive(Clone, ZeroizeOnDrop, Debug)]
 pub struct KeySharePrivate {
-    pub(crate) x: BigNumber, // in the range [1, q)
+    x: BigNumber, // in the range [1, q)
 }
 
-impl Debug for KeySharePrivate {
-    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
-        f.write_str("KeySharePrivate([redacted])")
+impl KeySharePrivate {
+    /// Get the private key share of the participant.
+    pub fn x(&self) -> BigNumber {
+        self.x.clone()
     }
-}
 
-impl KeySharePrivate {
-    // Computes the "raw" curve point corresponding to this private key.
+    /// Set x.
+    pub fn set_x(priv_share: BigNumber) -> KeySharePrivate {
```

Comment:
todo: it looks like `set_x` is always called immediately after generating the private share `x` (e.g. ,a random number mod q). I would recommend pulling the random number generation into the function and renaming it so the function is like
```
pub(crate) fn random(rng: &mut (impl CryptoRng + RngCore)) -> Self
```
That way, we never have the original copy of the randomly generated number floating around outside of this generation function.

Also note the reduced visibility of `pub(crate)`. We don't want external users to be able to create random private key shares at will. It's not a security issue exactly, because a random key share won't be attached to any meaningful key so it's not actually a secret. But it does help with misuse resistance, to avoid a confused caller accidentally making random key shares instead of using ones they got from previous protocol runs, for example.